Mathlib/CategoryTheory/Comma/OverClass.lean

```diff
@@ -108,7 +108,32 @@ instance [OverClass X S]
     [IsOverTower X S S'] [IsOverTower Y S S'] [HomIsOver f S] : HomIsOver f S' :=
   homIsOver_of_isOverTower f S S'
 
+variable (X) in
 /-- Bundle `X` with an `OverClass X S` instance into `Over S`. -/
 def OverClass.asOver [OverClass X S] : Over S := Over.mk (X ↘ S)
```

Comment:
This generates the lemma `OverClass.asOver_hom`:
```suggestion
@[simps! hom left]
def OverClass.asOver [OverClass X S] : Over S := Over.mk (X ↘ S)
```
In general, you may try `@[simps?]` or `@[simps!?]` and hovering over to see the exact form of the automatically generated simp lemmas.

Similarly, you can do `@[simps! left] def OverClass.asOverHom ...`.

Reply:
Change. Could you tell me why here it is safe to have `asOver_left` as a `simp` lemma? (I wonder if it causes `motive not correct` type issues)

Reply:
I am not sure I understand. Here, the simp lemmas are `dsimp` lemmas: they basically just unfold the definitions. We cannot always apply simp lemmas using the `rw` tactic (this may create `motive not correct` errors), but this does not happen with `simp` (and presumably `simp_rw`).

Reply:
For now I am fine with adding `asOver_left` with `@[simp]`. We will see if this causes trouble in this case, I am just trying to be cautious with these `simp` lemmas that rewrite types.